paramak/parametric_components/blanket_poloidal_segment.py

```diff
@@ -0,0 +1,316 @@
+                )
+                # create rectangle of dimension |AB|*2.6 x self.segments_gap
+                A = (inner_point[0], inner_point[1])
+                B = (outer_point[0], outer_point[1])
+
+                # increase rectangle length
+                security_factor = 0.8
+                local_thickness = distance_between_two_points(A, B)
+                A = extend(A, B, -local_thickness * security_factor)
+                B = extend(A, B, local_thickness * (1 + 2 * security_factor))
+                # create points for cutter
+                points_cutter = [
+                    A,
+                    B,
+                    rotate(
+                        B,
+                        extend(
+                            B,
+                            A,
+                            self.segments_gap),
+                        angle=-np.pi / 2),
+                    rotate(
+                        A,
+                        extend(
+                            A,
+                            B,
+                            self.segments_gap),
+                        angle=np.pi / 2)]
+                cutter.points = points_cutter
+                # add cutter to global cutting shape
+                cutting_shape.union.append(cutter)
+
+            self.segments_cutters = cutting_shape
+
+
+def compute_lengths_from_angles(angles, distribution):
+    """Computes the length of segments between a set of points on a (x,y)
+    distribution.
+
+    Args:
+        angles (list): Contains the angles of the points (degree)
+        distribution (callable): function taking an angle as argument and
+            returning (x,y) coordinates.
+
+    Returns:
+        list: contains the lengths of the segments.
+    """
+    points = []
+    for angle in angles:
+        points.append(distribution(angle))
+
+    lengths = []
+    for i in range(len(points) - 1):
+        lengths.append(distance_between_two_points(points[i], points[i + 1]))
+    return lengths
+
+
+def segments_optimiser(length_limits, nb_segments_limits, distribution, angles,
+                       stop_on_success=True):
+    """Optimiser segmenting a given R(theta), Z(theta) distribution of points
+    with constraints regarding the number of segments and the length of the
+    segments.
+
+    Args:
+        length_limits ((float, float)): The minimum and maximum acceptable
+            length of the segments. Ex: (100, 500), (100, None), (None, 300),
+            None, (None, None)
+        nb_segments_limits ((int, int)): The minimum and maximum acceptable
+            number of segments. Ex: (3, 10), (5, None), (None, 7),
+            None, (None, None)
+        distribution (callable): function taking an angle as argument and
+            returning (x,y) coordinates.
+        angles ((float, float)): the start and stop angles of the distribution.
+        stop_on_sucess (bool, optional): If set to True, the optimiser will
```

Comment:
perhaps the user doesn't need to set stop_on_success, perhaps it should always be true

Reply:
Done